### PR TITLE
Update physiofit to v3.3.2

### PR DESCRIPTION
**Pull request by llegregam**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-metabolomics/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by llegregam** [comment]
I do not understand, the planemo tests pass on my system but not on the CI instance...

**Comment by bgruening** [comment]
They are 2 failing tests here. One is a linting error that you probably do not run locally. Your R code is not complying to the best-practices of R code.

The second failure is the tool test failing. I don't know why, but you can try to run the tests locally with `planemo test --biocontainers ....`. This will run at least the tests exactly as here on CI.

**Comment by llegregam** [comment]
When I run the linter I get no errors on my wrapper, and my tool doesn't implement any R code so I don't understand what you mean.
When I try running the tests using --biocontainers, I get this exception: "Exception: Failed to install involucro for Planemo.". Is it my test env that is not well set up? I'm using planemo on an ubuntu VM.
Regards

File: tools/physiofit/physiofit.xml [hunk 1]
```diff
@@ -1,278 +1,98 @@
...
-    <data name='stats' label='Stat_results' format='tabular'/>
-    <data name='config' label='Config_file' format='json'/>
-    <data name='log' label='Run_Log' format='txt'/>
+        <!-- <data name='config' label='Config_file' format='yaml'/> -->
+        <data name='recap' label='Summary file' format='csv'/>
+        <data name='zip' label='Results' format='zip'/>
+        <data name='log' label='Log' format='txt'/>
+        <!-- <collection name='results' type='list' label='Resulting files'>
+            <discover_datasets pattern='__name_and_ext__'/>
+        </collection> -->
```
Comment (lecorguille) [inline tools/physiofit/physiofit.xml:24]: The collection seems a better solution than zip, why give up?

File: tools/physiofit/physiofit.xml [hunk 2]
```diff
@@ -1,278 +1,98 @@
...
-                    <param name='flux_biom_bounds' type='text' value='()' label='Bounds on biomass fluxes. Bounds should be given in tuple format (see docs for more information)' />
-                </when>
-            </conditional>
-
-            <param name='verbose' type='boolean' argument='--verbose' truevalue='-v' falsevalue='' label='Flag to get debug information'/>
-
-        </section>
-
+        <param name='datafile' type='data' format='tabular, txt' multiple='true' label='Upload data file containing biomass and concentration values'/>
+        <param name='configfile' type='data' format='yaml' multiple='true' label='Upload yaml configuration file containing run parameters'/>
```
Comment (lecorguille) [inline tools/physiofit/physiofit.xml:15]: Can't you provide the different `<param>` to the user and template the configfile?

**Comment by lecorguille** [comment]
> They are 2 failing tests here. One is a linting error that you probably do not run locally. Your R code is not complying to the best-practices of R code.

When in doubt, reboot
I just relaunched the step and it succeeded ✨

File: tools/physiofit/physiofit.xml (hunk 1 above)
Comment (llegregam) [reply tools/physiofit/physiofit.xml:24]: I tried to but the discover_datasets is not able to detect the generated files!

File: tools/physiofit/physiofit.xml (hunk 2 above)
Comment (llegregam) [reply tools/physiofit/physiofit.xml:15]: No depending on the selected models and such the config file changes and as such it is simpler to have the user fill in the different run parameters beforehand

**Comment by llegregam** [comment]
I have tried switching up the tests and testing in biocontainers on my own system, but impossible to understand the source of the errors. Without biocontainers all tests pass but as soon as I switch back to containers there is no making it work. Any ideas?

**Comment by bgruening** [comment]
This is what I'm seeing:

grafik

The tool is throwing an error code of 2 :(

You can also run `planemo serve --biocontainers ` and try it interactively, maybe you see something?

Another thing you can try is adding `--no_cleanup` this will leave the job working directory and you can investigate whats happening outside of Galaxy.

**Comment by llegregam** [comment]
So after a lot of coercion I was able to get a clean serve inside the biocontainers (with the right tool version) and was able to get a log from my tool. Here is the result:
image
Apparently we cannot create files during job execution? Since when?

**Comment by llegregam** [comment]
Fixed the bug, see #256 for more information!

**Comment by llegregam** [comment]
@bgruening @lecorguille @yguitton 
Could I have a quick review please? :)

**Comment by yguitton** [comment]
Hi @llegregam 
I think i can accept the PR
Yann

**Pull request merged**